src/engine/GroupBy.cpp

```diff
@@ -1256,13 +1322,20 @@ void GroupBy::computeGroupByForHashMapOptimization(
     evaluationContext._beginIndex = i;
     evaluationContext._endIndex = std::min(i + blockSize, subresult.size());
 
-    auto currentBlockSize =
-        evaluationContext._endIndex - evaluationContext._beginIndex;
+    auto currentBlockSize = evaluationContext.size();
 
     // Perform HashMap lookup once for all groups in current block
-    auto groupValues =
-        subresult.getColumn(columnIndex)
-            .subspan(evaluationContext._beginIndex, currentBlockSize);
+    using U = HashMapAggregationData<GROUPED_COLUMNS>::template T<
+        std::span<const Id>>;
+    U groupValues;
+    if constexpr (requires { groupValues.resize(columnIndices.size()); })
+      groupValues.resize(columnIndices.size());
```

Comment:
You could make this a small local (static or so) helper function `resizeIfVector`, then you have this rather technical piece of code out of your actual logic.